### PR TITLE
feat: replace python-linter with mypy specific action

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

File: .github/workflows/partial-tests.yml [hunk 1]
```diff
@@ -9,14 +9,9 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - uses: ricardochaves/python-lint@v1.4.0
+      - uses: jpetrucciani/mypy-check@master
```
Comment (trgiangdo) [inline .github/workflows/partial-tests.yml:12]: This action uses mypy 1.9.0. I like it :D

**Review by trgiangdo — COMMENTED** [review]
The new action raises quite a number of new error. We need to handle these first before merging

File: .github/workflows/partial-tests.yml (hunk 1 above)
Comment (joaoandre-avaiga) [reply .github/workflows/partial-tests.yml:12]: Yeah, I chose this one because it updates quite often. Now to fix the erros so the linter passes

**Review by trgiangdo — APPROVED** [review]
LGTM

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**